### PR TITLE
[special] Alias for sigmoid and logit & follow-up

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: #50345

Chages:
* Alias for sigmoid and logit
* Adds out variant for C++ API
* Updates docs to link back to `special` documentation

### CODE REVIEW

**Comment by ezyang** [comment]
ACKing `native_functions.yaml` changes.

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -7688,25 +7598,8 @@ def merge_dicts(*dicts):
...
-
-Example::
-
-    >>> a = torch.randn(4)
-    >>> a
-    tensor([ 0.9213,  1.0887, -0.8858, -1.7683])
-    >>> torch.sigmoid(a)
-    tensor([ 0.7153,  0.7481,  0.2920,  0.1458])
-""".format(**common_args))
+Alias for :func:`torch.special.expit`.
```
Comment (mruberry) [inline torch/_torch_docs.py:7601]: This is OK for this PR, but ahead of the 1.9 release we should consider if "expit" is our preferred name for this operation.

File: torch/csrc/jit/passes/normalize_ops.cpp [hunk 2]
```diff
@@ -90,6 +90,7 @@ const std::unordered_map<Symbol, Symbol>& getOperatorAliasMap() {
       {aten::special_erf, aten::erf},
       {aten::special_erfc, aten::erfc},
       {aten::special_erfinv, aten::erfinv},
+      {aten::special_expit, aten::sigmoid},
```
Comment (mruberry) [inline torch/csrc/jit/passes/normalize_ops.cpp:93]: What about special.logit?

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -3861,6 +3862,16 @@ def reference_lgamma(x):
                    dtypesIfCUDA=all_types_and(torch.bool, torch.half, torch.bfloat16),
                    sample_inputs_func=sample_inputs_logit,
                    safe_casts_outputs=True),
+    UnaryUfuncInfo('special.logit',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3865]: Should this be an OpInfo for torch.logit with special.logit as an alias?

**Review by mruberry — COMMENTED** [review]
Hey @kshitij12345! Overall this looks great, like usual, just a couple small questions.

File: torch/csrc/jit/passes/normalize_ops.cpp (hunk 2 above)
Comment (kshitij12345) [reply torch/csrc/jit/passes/normalize_ops.cpp:93]: For logit, I decided to stick with the signature of `scipy.special.logit` which doesn't have `eps` argument.
Thus `special_logit` is not actually an alias to `logit` as the signature is different.
Let me know what you feel about the same.

Reference: 
https://pytorch.org/docs/stable/generated/torch.logit.html
https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.logit.html

Comment (mruberry) [reply torch/csrc/jit/passes/normalize_ops.cpp:93]: I would just add the eps. It's pretty common when implementing a NumPy (or SciPy) function that it's missing some functionality from the current Torch version, and the only way we can really tell people to use the new thing is if we extend it. Since this functionality is a superset of the current special.logit functionality there's no compatibility concern with adding it.

**Comment by kshitij12345** [comment]
Gentle Ping :)

File: torch/special/__init__.py [hunk 4]
```diff
@@ -118,10 +118,67 @@
...
+    z_{i} = \begin{cases}
+        x_{i} & \text{if eps is None} \\
+        \text{eps} & \text{if } x_{i} < \text{eps} \\
+        x_{i} & \text{if } \text{eps} \leq x_{i} \leq 1 - \text{eps} \\
+        1 - \text{eps} & \text{if } x_{i} > 1 - \text{eps}
+    \end{cases}
+""" + r"""
+Args:
+    {input}
+    eps (float, optional): the epsilon for input clamp bound. Default: ``None``
```
Comment (mruberry) [inline torch/special/__init__.py:144]: This language is consistent with what we have now, but ahead of 1.9 we probably want to do a docs review to tweak it

**Review by mruberry — APPROVED** [review]
LGTM!

Would you take a note that we should do a torch.special docs review ahead of the PyTorch 1.9 branch cut, too? We did the same thing with torch.linalg ahead of the PyTorch 1.8 branch cut and plan to do another torch.linalg docs review for PyTorch 1.9, too.

File: torch/special/__init__.py (hunk 4 above)
Comment (kshitij12345) [reply torch/special/__init__.py:144]: Noted. :+1:
